prototype/mobile_endpoint/views/receiver.py

```diff
@@ -25,6 +26,7 @@ def couch_receiver(domain):
 @ota_mod.route('/mongo-receiver/<domain>', methods=['POST'])
 @requires_auth
 def mongo_receiver(domain):
+    connect(host=current_app.config.get('MONGO_URI'))
```

Comment:
Can you not do it in `mobile_endpoint.__init__.create_app` ?

On 6 August 2015 at 21:30, Noah Carnahan <notifications@github.com> wrote:

> In prototype/mobile_endpoint/views/receiver.py
> <https://github.com/dimagi/mobile-endpoint/pull/45#discussion_r36455430>:
>
> > @@ -25,6 +26,7 @@ def couch_receiver(domain):
> >  @ota_mod.route('/mongo-receiver/<domain>', methods=['POST'])
> >  @requires_auth
> >  def mongo_receiver(domain):
> > +    connect(host=current_app.config.get('MONGO_URI'))
>
> Initiating the mongoengine connection here feels wrong, but didn't spend
> the time to figure out the "right" place to put this. (Docs seem to suggest
> at the top of the models file, but flask.current_app isn't always known
> when that is first imported...)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dimagi/mobile-endpoint/pull/45/files#r36455430>.
>



-- 
Simon Kelly
Senior Engineer | Dimagi South Africa